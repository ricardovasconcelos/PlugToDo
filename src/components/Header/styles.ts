import styled from 'styled-components';

export const HeaderContainer = styled.header`
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 3.75rem;

  img {
    margin-left: 1rem;
  }

  a {
    &:focus {
      box-shadow: none;
    }
  }
`;
