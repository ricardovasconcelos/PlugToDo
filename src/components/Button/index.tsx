import { ButtonHTMLAttributes, ReactNode } from 'react';

import { ButtonComponent } from './styles';

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  children: ReactNode;
  width: number;
}

export function Button({ children, ...props }: ButtonProps) {
  return <ButtonComponent {...props}>{children}</ButtonComponent>;
}
