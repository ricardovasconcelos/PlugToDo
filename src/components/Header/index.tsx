import { HeaderContainer } from './styles';
import logo from '../../assets/logo.svg';
import { Link } from 'react-router-dom';

export function Header() {
  return (
    <HeaderContainer>
      <Link to='/'>
        <img src={logo} alt='Plug Logo' />
      </Link>
    </HeaderContainer>
  );
}
