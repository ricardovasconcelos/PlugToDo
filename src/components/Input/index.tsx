import { InputHTMLAttributes } from 'react';

import { InputComponent } from './styles';

interface InputProps extends InputHTMLAttributes<HTMLInputElement> {}

export function Input({ ...props }: InputProps) {
  return <InputComponent {...props} />;
}
