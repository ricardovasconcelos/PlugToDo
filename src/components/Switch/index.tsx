import { InputHTMLAttributes } from 'react';
import { SwithContainer, SwitchWrapper, SwitchCheckbox, SwitchItem } from './styles';

interface SwitchProps extends InputHTMLAttributes<HTMLInputElement> {
  label: string;
  checked?: boolean;
}

export function Switch({ label, checked, ...props }: SwitchProps) {
  return (
    <SwithContainer data-testid='switch-container'>
      <p>{label}</p>

      <SwitchWrapper>
        <SwitchCheckbox id='checkbox' type='checkbox' checked={checked} {...props} />
        <SwitchItem htmlFor='checkbox' />
      </SwitchWrapper>
    </SwithContainer>
  );
}
