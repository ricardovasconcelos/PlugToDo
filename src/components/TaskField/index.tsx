import { TaskFieldContainer, TaskDetailContainer } from './styles';
import { Task } from '../../db/db.types';

interface TaskFieldProps extends Omit<Task, 'id'> {
  onClickTask: () => void;
}

export function TaskField({ title, description, done, onClickTask }: TaskFieldProps) {
  return (
    <TaskFieldContainer done={done} onClick={onClickTask}>
      <TaskDetailContainer>
        <h3>{title}</h3>
        <p>{description}</p>
      </TaskDetailContainer>
    </TaskFieldContainer>
  );
}
