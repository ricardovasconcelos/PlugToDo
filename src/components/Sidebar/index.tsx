import { useRef, useEffect, useCallback, MouseEvent } from 'react';
import { useSpring, animated } from 'react-spring';
import { Button } from '../Button';
import { Input } from '../Input';

import {
  SidebarContainer,
  SidebarWrapper,
  SidebarContent,
  SidebarHeader,
  CloseSidebar,
  SidebarFooter,
} from './styles';

import close from '../../assets/close.svg';

interface SidebarProps {
  showSidebar: boolean;
  onShowSidebar: (value: boolean) => void;
}

export const Sidebar = ({ showSidebar, onShowSidebar }: SidebarProps) => {
  const sidebarRef = useRef<HTMLDivElement>(null);

  const animation = useSpring({
    config: {
      duration: 250,
    },
    opacity: showSidebar ? 1 : 0,
    transform: showSidebar ? `translateY(0%)` : `translateY(-100%)`,
  });

  const closeSidebar = (event: MouseEvent<HTMLDivElement>) => {
    if (sidebarRef.current === event.target) {
      onShowSidebar(false);
    }
  };

  const keyPress = useCallback(
    (event: KeyboardEvent) => {
      if (event.key === 'Escape' && showSidebar) {
        onShowSidebar(false);
      }
    },
    [onShowSidebar, showSidebar],
  );

  useEffect(() => {
    document.addEventListener('keydown', keyPress);
    return () => document.removeEventListener('keydown', keyPress);
  }, [keyPress]);

  return (
    <>
      {showSidebar ? (
        <SidebarContainer onClick={closeSidebar} ref={sidebarRef}>
          <animated.div style={animation}>
            <SidebarWrapper>
              <SidebarHeader>
                <CloseSidebar onClick={() => onShowSidebar(false)}>
                  <img src={close} alt='Botão para fechar modal de tarefas' />
                </CloseSidebar>
                <p>Nova Tarefa</p>
              </SidebarHeader>

              <SidebarContent>
                <Input placeholder='Título' />

                <Input placeholder='Descrição' />
              </SidebarContent>

              <SidebarFooter>
                <Button width={20}>CRIAR TAREFA</Button>
              </SidebarFooter>
            </SidebarWrapper>
          </animated.div>
        </SidebarContainer>
      ) : null}
    </>
  );
};
