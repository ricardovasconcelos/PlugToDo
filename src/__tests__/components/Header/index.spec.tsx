import { screen, render } from '@testing-library/react';

import { Header } from '../../../components/Header';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Header', () => {
  it('should be able to render a Header', () => {
    const { container } = render(
      <Router>
        <Header />
      </Router>,
    );
    expect(screen.getByRole('banner')).toHaveStyle({
      height: '3.75rem',
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'space-between',
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
