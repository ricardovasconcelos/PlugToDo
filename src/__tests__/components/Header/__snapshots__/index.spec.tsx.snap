// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should be able to render a Header 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 3.75rem;
}

.c0 img {
  margin-left: 1rem;
}

.c0 a:focus {
  box-shadow: none;
}

<header
  class="c0"
>
  <a
    href="/"
  >
    <img
      alt="Plug Logo"
      src="SvgrURL"
    />
  </a>
</header>
`;
